### PR TITLE
CEDS-1908 Fix for changing countries

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/controllers/declaration/RoutingCountriesSummaryController.scala [hunk 1]
```diff
@@ -122,7 +122,7 @@ class RoutingCountriesSummaryController @Inject()(
 
   def submitChangeCountry(mode: Mode, countryToChange: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
     val cachedCountries = request.cacheModel.locations.routingCountries
-    val countriesForValidation = cachedCountries.filter(_ == countryToChange)
+    val countriesForValidation = cachedCountries.filterNot(_ == countryToChange)
```
Comment (SteveSugden) [inline app/controllers/declaration/RoutingCountriesSummaryController.scala]: So countriesForValidation are the existing countries in the cache except the one you want to change.  Makes sense.

Comment (PatrykRudnicki) [reply app/controllers/declaration/RoutingCountriesSummaryController.scala]: Yes, to check duplication in easier way

File: test/unit/controllers/declaration/RoutingCountriesSummaryControllerSpec.scala [hunk 2]
```diff
@@ -281,4 +281,18 @@ class RoutingCountriesSummaryControllerSpec extends ControllerSpec {
...
+
+  "Routing countries controller" should {
+
+    "handle duplication during changing the country" in {
+
+      withNewCaching(aDeclaration(withRoutingCountries(Seq("PL", "DZ", "AD"))))
+
+      val duplicatedCountry = JsObject(Seq("country" -> JsString("PL")))
+
+      val result = controller.submitChangeCountry(Mode.Normal, "DZ")(postRequest(duplicatedCountry))
```
Comment (SteveSugden) [inline test/unit/controllers/declaration/RoutingCountriesSummaryControllerSpec.scala]: So in this request, DZ is the country I want to change, and I want to change it to PL (which is already in the list - hence error).  Is that right?

Comment (PatrykRudnicki) [reply test/unit/controllers/declaration/RoutingCountriesSummaryControllerSpec.scala]: Yes, exactly. I can add some local variables to describe those countries if you want

**Pull request merged**